gallery/gallery-war/src/main/java/org/silverpeas/components/gallery/web/AbstractGalleryResource.java

```diff
@@ -268,9 +370,9 @@ protected View getMediaEmbed(final MediaType expectedMediaType, final String med
    * @param mediaId the media identifier
    * @param thumbnailId the thumbnail identifier
    * @param sizeDirective the size directive with pattern (optional)
-   * @return
+   * @return the response.
    */
-  protected Response getMediaThumbnail(final MediaType expectedMediaType, final String mediaId,
+  Response getMediaThumbnail(final MediaType expectedMediaType, final String mediaId,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "getMediaThumbnail" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)